### PR TITLE
Add utests and stests for control interface authorization

### PR DESCRIPTION
**Pull request by christoph-hamm**
Issues: #22

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: tests/stests/control_interface/authorization.robot [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+*** Variables ***
+${simple_yaml_file}      ${EMPTY}
+${manifest12_yaml_file}  ${EMPTY}
+${manifest1_yaml_file}   ${EMPTY}
+${manifest2_yaml_file}   ${EMPTY}
+${manifest_no_agent_name_yaml_file}    ${EMPTY}
+
+*** Test Cases ***
+
+No rules
```
Comment (lingnoi) [inline tests/stests/control_interface/authorization.robot:32]: As we mentioned BDT in our [contribution guidelines](https://eclipse-ankaios.github.io/ankaios/latest/development/system-tests/#behavior-driven-system-test), we should write the system tests in that way. But If we decide to not to be that strict, then we should mention this in the contribution guidelines!

**Review by lingnoi — CHANGES_REQUESTED** [review]
Just some small findings.

File: agent/src/control_interface/authorizer.rs [hunk 2]
```diff
@@ -192,264 +193,229 @@ impl From<&ControlInterfaceAccess> for Authorizer {
...
-            sections: if value.is_empty() {
-                Vec::new()
-            } else {
-                value.split(PATH_SEPARATOR).map(Into::into).collect()
-            },
-        }
+    #[derive(Clone, Debug, PartialEq)]
+    pub struct MockRule<T> {
+        patterns: Option<Vec<T>>,
+        phantom: PhantomData<T>,
```
[the remark this answers is not in the document]
Comment (christoph-hamm) [reply agent/src/control_interface/authorizer.rs]: Removed.

File: tests/stests/control_interface/authorization.robot (hunk 1 above)
Comment (christoph-hamm) [reply tests/stests/control_interface/authorization.robot:35]: I have changed the stests to follow the BDT style.

File: tests/stests/control_interface/authorization.robot [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    And the controller workload updates the state with manifest "${CONFIGS_DIR}/simple_state.yaml" and update mask desiredState.workloads.simple_existing.tags
+    And the controller workload gets the state
+    And the controller workload gets the state of fields desiredState.workloads.simple_existing
+
+    Then the controller workload requests shall all fail
+
+
+Allow read write rule for workloads except write to simple_existing allows all read and write on workloads execept write to simple_existing
+    Given the controller workload is allowed to read and write on desiredState.workloads
+    Given the controller workload is forbidden to to write on desiredState.workloads.simple_existing
```
Comment (lingnoi) [inline tests/stests/control_interface/authorization.robot:129]:
```suggestion
    And the controller workload is forbidden to to write on desiredState.workloads.simple_existing
```

File: tests/stests/control_interface/authorization.robot [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    When the controller workload updates the state with manifest "${CONFIGS_DIR}/simple_state.yaml" and update mask desiredState.workloads.simple
+    And the controller workload gets the state of fields desiredState.workloads.simple_existing
+    And the controller workload gets the state of fields desiredState.workloads.simple_existing.tags
+
+    Then the controller workload requests shall all succeed
+
+Allow read write rule for workloads except write to simple_existing forbids writing to simple_existing and reading whole state
+    Given the controller workload is allowed to read and write on desiredState.workloads
+    Given the controller workload is forbidden to to write on desiredState.workloads.simple_existing
```
Comment (lingnoi) [inline tests/stests/control_interface/authorization.robot:139]:
```suggestion
    And the controller workload is forbidden to to write on desiredState.workloads.simple_existing
```

**Review by inf17101 — APPROVED** [review]
LGTM 👍

**Pull request merged**